### PR TITLE
[ARCTIC-1323] Extract the toString() logic of ArcticSplit to the core module

### PR DESCRIPTION
**Pull request by YesOrNo828**
## Why are the changes needed?

ArcticAplit class in the flink modules already has a toString method to display the metrics about the files/tasks of Arctic planning.

We could abstract the common toString to the core module, and display information about the NodeFileScanTask class.

Fixed: #1323 

```java
LOG.info(baseCombinedScanTask.toString()):

BaseCombinedScanTask{
tasks=NodeFileScanTask{
	baseTasks=[], 
	insertTasks=[DefaultKeyedFile{
		file=/tmp/test.parquet, 
		type=I, 
		mask=0, 
		index=0, 
		transactionId=2, 
		fileSizeInBytes=10, 
		recordCount=100}], 
	deleteFiles=[]}}
```

## Brief change log

  - *Extract the toString() logic of ArcticSplit into the core module*
  - *The flink 1.12, 1.14, and 1.15 versions are affected.*
  - *Add UTs*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/scan/BaseCombinedScanTask.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public Collection<KeyedTableScanTask> tasks() {
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
-        .add("tasks", Joiner.on(", ").join(tasks))
+        .add("\ntasks", Joiner.on(",\n").join(tasks).replaceAll("\n", "\n\t"))
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/scan/BaseCombinedScanTask.java:53]: Can the statement `Joiner.on(",\n").join(tasks).replaceAll("\n", "\n\t")` be simplified to `Joiner.on(",\n\t").join(tasks)` ?

**Review by zhoujinsong — COMMENTED** [review]
@YesOrNo828 
Can you put some sample of logs to help us understand the log format.

File: core/src/main/java/com/netease/arctic/scan/BaseCombinedScanTask.java (hunk 1 above)
Comment (YesOrNo828) [reply core/src/main/java/com/netease/arctic/scan/BaseCombinedScanTask.java:53]: After testing and displaying the logs, I would prefer the prefix format:
```
MoreObjects.toStringHelper(this)
        .add("\ntasks", Joiner.on(",\n").join(tasks).replaceAll("\n", "\n\t"))
        .toString()

BaseCombinedScanTask{
tasks=NodeFileScanTask{
	baseTasks=[], 
	insertTasks=[DefaultKeyedFile{
		file=/tmp/test.parquet, 
		type=I, 
		mask=0, 
		index=0, 
		transactionId=2, 
		fileSizeInBytes=10, 
		recordCount=100}], 
	deleteFiles=[]}}

----
MoreObjects.toStringHelper(this)
        .add("\ntasks", Joiner.on(",\n\t").join(tasks))
        .toString();


BaseCombinedScanTask{
tasks=NodeFileScanTask{
baseTasks=[], 
insertTasks=[DefaultKeyedFile{
	file=/tmp/test.parquet, 
	type=I, 
	mask=0, 
	index=0, 
	transactionId=2, 
	fileSizeInBytes=10, 
	recordCount=100}], 
deleteFiles=[]}}
```

**Review by zhoujinsong — APPROVED** [review]
LGTM.
